### PR TITLE
Enable numa_memory on ppc.

### PR DESCRIPTION
**Pull request by harihare**
Enable code to run the tests on PPC with the same config. The
numa numbers in the config will act as the index value for the
numa list for PPC.

Signed-off-by: Hariharan T.S <hari@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by harihare** [comment]
**### Test Log with Patch**
# python avocado-setup.py --run-suite guest_numa --guest-os RHEL.7.6alt.ppc64le --only-filter 'virtio_scsi virtio_net qcow2' --no-download
12:01:27 INFO    : Check for environment
12:01:43 INFO    : 
12:01:43 INFO    : Running Guest Tests Suite numa
12:01:43 INFO    : Running: /usr/bin/avocado run --vt-type libvirt --vt-config /home/hariharan/avocado-fvt-wrapper/data/avocado-vt/backends/libvirt/cfg/numa.cfg                 --force-job-id 9fe27a58aef48f4c33c9a26841556da4e8accbd6  --vt-only-filter                                             "virtio_scsi virtio_net qcow2 RHEL.7.6alt.ppc64le" --job-results-dir /home/hariharan/avocado-fvt-wrapper/results
JOB ID     : 9fe27a58aef48f4c33c9a26841556da4e8accbd6
JOB LOG    : /home/hariharan/avocado-fvt-wrapper/results/job-2019-01-30T12.01-9fe27a5/job.log
 (01/24) guest_import_32G.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-qemu.unattended_install.import.import.default_install.aio_native: PASS (60.04 s)
 (02/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.possitive_test.no_vcpu.mem_auto.strict: PASS (48.89 s)
 (03/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.possitive_test.no_vcpu.mem_auto.interleave: PASS (50.26 s)
 (04/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.possitive_test.no_vcpu.mem_nodeset.node1.strict: PASS (54.42 s)
 (05/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.possitive_test.no_vcpu.mem_nodeset.node1.preferred: PASS (46.87 s)
 (06/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.possitive_test.no_vcpu.mem_nodeset.node2.strict: PASS (53.74 s)
 (07/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.possitive_test.no_vcpu.mem_nodeset.node2.interleave: PASS (54.14 s)
 (08/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.possitive_test.vcpu.vcpu_auto.mem_auto.strict: PASS (56.54 s)
 (09/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.possitive_test.vcpu.vcpu_auto.mem_auto.interleave: PASS (56.37 s)
 (10/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.possitive_test.vcpu.vcpu_auto.mem_nodeset.node1.strict: PASS (56.57 s)
 (11/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.possitive_test.vcpu.vcpu_auto.mem_nodeset.node1.preferred: PASS (56.41 s)
 (12/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.possitive_test.vcpu.vcpu_auto.mem_nodeset.node2.strict: PASS (56.49 s)
 (13/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.possitive_test.vcpu.vcpu_auto.mem_nodeset.node2.interleave: PASS (56.20 s)
 (14/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.possitive_test.vcpu.vcpu_static.mem_auto.strict: PASS (56.87 s)
 (15/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.possitive_test.vcpu.vcpu_static.mem_auto.interleave: PASS (56.49 s)
 (16/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.possitive_test.vcpu.vcpu_static.mem_nodeset.node1.strict: PASS (54.45 s)
 (17/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.possitive_test.vcpu.vcpu_static.mem_nodeset.node1.preferred: PASS (49.95 s)
 (18/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.possitive_test.vcpu.vcpu_static.mem_nodeset.node2.strict: PASS (49.75 s)
 (19/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.possitive_test.vcpu.vcpu_static.mem_nodeset.node2.interleave: PASS (49.53 s)
 (20/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.negative_test.out_range.strict: PASS (14.88 s)
 (21/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.negative_test.out_range.preferred: PASS (13.74 s)
 (22/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.negative_test.out_range.interleave: PASS (13.79 s)
 (23/24) guest_numa_memory.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.numa_memory.negative_test.preferred_multi: PASS (13.63 s)
 (24/24) guest_remove_32G.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.RHEL.7.6alt.ppc64le.powerkvm-libvirt.remove_guest.without_disk: PASS (6.32 s)
RESULTS    : PASS 24 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1097.55 s
JOB HTML   : /home/hariharan/avocado-fvt-wrapper/results/job-2019-01-30T12.01-9fe27a5/results.html
12:20:02 INFO    : 
12:20:02 INFO    : Summary of test results can be found below:
TestSuite                                                                   TestRun    Summary             
 
guest_numa                                                                  Run        Successfully executed
/home/hariharan/avocado-fvt-wrapper/results/job-2019-01-30T12.01-9fe27a5/job.log
12:20:02 INFO    : Removing temporary mux dir

File: libvirt/tests/src/numa/numa_memory.py [hunk 1]
```diff
@@ -109,13 +110,35 @@ def numa_mode_check(mode_nodeset):
     vm = env.get_vm(vm_name)
     backup_xml = libvirt_xml.VMXML.new_from_dumpxml(vm_name)
 
+    # Get host numa node list
+    host_numa_node = utils_misc.NumaInfo()
+    node_list = host_numa_node.online_nodes_withmem
+    logging.debug("host node list is %s", node_list)
```
Comment (balamuruhans) [inline libvirt/tests/src/numa/numa_memory.py]: for better readability to print list by " ".join(node_list)

File: libvirt/tests/src/numa/numa_memory.py [hunk 2]
```diff
@@ -109,13 +110,35 @@ def numa_mode_check(mode_nodeset):
...
         value = params.get(mem_param)
         if value:
             numa_memory[mem_param.split('_')[1]] = value
+    arch = platform.machine()
+    if 'ppc64' in arch:
+        try:
+            ppc_memory_nodeset = ""
+            nodes = numa_memory["nodeset"]
+            if '-' in nodes:
+                for n in range(int(nodes.split('-')[0]), int(nodes.split('-')[1]) + 1):
```
Comment (balamuruhans) [inline libvirt/tests/src/numa/numa_memory.py]: appropriate variable name can be used

File: libvirt/tests/src/numa/numa_memory.py [hunk 3]
```diff
@@ -109,13 +110,35 @@ def numa_mode_check(mode_nodeset):
...
+                    ppc_memory_nodeset += str(node_list[n]) + ','
+            else:
+                node_lst = nodes.split(',')
+                for n in range(len(node_lst)):
```
Comment (balamuruhans) [inline libvirt/tests/src/numa/numa_memory.py]: same here

File: libvirt/tests/src/numa/numa_memory.py [hunk 4]
```diff
@@ -109,13 +110,35 @@ def numa_mode_check(mode_nodeset):
...
+                    ppc_memory_nodeset += str(node_list[int(node_lst[n])]) + ','
+            numa_memory["nodeset"] = ppc_memory_nodeset[:-1]
+        except IndexError:
```
Comment (balamuruhans) [inline libvirt/tests/src/numa/numa_memory.py]:
```python
except (KeyError, IndexError):
    pass
```

File: libvirt/tests/src/numa/numa_memory.py (hunk 2 above)
Comment (harihare) [reply libvirt/tests/src/numa/numa_memory.py]: Replace n with nnode (numanode)

File: libvirt/tests/src/numa/numa_memory.py (hunk 1 above)
Comment (harihare) [reply libvirt/tests/src/numa/numa_memory.py]: Updated as mentioned

**Comment by harihare** [comment]
All review comments resolved.

**Pull request merged**